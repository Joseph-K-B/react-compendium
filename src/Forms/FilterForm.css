.filterForm {
    display: flex;
    flex-direction: row;
    text-align: center;
    /* align-items: center; */
    width: 100%;
    height: 100%;
    /* border: solid red 0.25rem; */
}

.sortType {
    position: relative;
    display: flex;
    flex-direction: column;
    text-align: center;
    margin: 5%;
}

.sortType select {
    color: #D3D0CB;
    background-color: #393E41;
    box-shadow: 0 0 5px 3px red;
    text-align: center;
    height: 2rem;
    width: auto;
    -webkit-appearance: none;
}

.sortType select:hover {
    color: black;
    background-color: #D3D0CB;
}



/* .pageBtn {
    max-height: 3rem;
    max-width: 5rem;
    margin: 5%;
} */

