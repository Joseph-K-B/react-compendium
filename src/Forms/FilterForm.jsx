import './FilterForm.css';

function FilterForm({    
    types, 
    selectedType, 
    filterChange,
}) {
    return (
        <section className='filterForm'>
            <div aria-label='sortType' className='sortType'>            
            <h3>Type</h3>
            <select
                className='dropDown'
                value={selectedType}
                onChange={(e) => filterChange(e.target.value)}    
            >
                    <option value='all' key='all'>All</option>
                    {types.map(({type}) => (
                        <option value={type} key={type}>{type}</option>
                    ))}
            </select>
            </div>                
        </section>            
    );
};

export default FilterForm;