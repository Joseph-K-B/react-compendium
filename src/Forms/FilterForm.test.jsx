import { render, screen } from "@testing-library/react";
import FilterForm from "./FilterForm"


it('should render form controls', () => {
    const {container} = render(<FilterForm types={['fire']}/>);
    const sortType = screen.getByLabelText(/sortType/);
    const sortOrder = screen.getByLabelText(/sortOrder/);
    const sortStat = screen.getByLabelText(/sortStat/);

    expect(sortType).toBeInTheDocument();
    expect(sortOrder).toBeInTheDocument();
    expect(sortStat).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})