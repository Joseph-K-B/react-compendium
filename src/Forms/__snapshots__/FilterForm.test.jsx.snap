// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render filter form controls 1`] = `
<div>
  <section
    class="filterForm"
  >
    <div
      aria-label="sortType"
      class="sortType"
    >
      <h3>
        Type
      </h3>
      <select
        class="dropDown"
      >
        <option
          value="all"
        >
          All
        </option>
        <option />
      </select>
    </div>
  </section>
</div>
`;
